broker/cloud_run/lsst/lite/main.py

```diff
@@ -0,0 +1,170 @@
+    try:
+        alert = pittgoogle.Alert.from_cloud_run(envelope, f"{SURVEY}")
+    except pittgoogle.exceptions.BadRequest as exc:
+        return str(exc), HTTP_400
+
+    TOPIC_LITE.publish(_create_lite_alert(alert), serializer="json")
+
+    return "", HTTP_204
+
+
+def _create_lite_alert(alert: pittgoogle.Alert) -> pittgoogle.Alert:
+    """Create a "lite" alert containing a subset of the fields of the original alert packet."""
+
+    # create dictionaries
+    object_lite_dict = _create_lite_dict(
+        alert.dict.get(alert.get_key("object")), _create_object_fields_list(alert)
+    )
+    prev_sources_lite_dict = _create_prv_sources_dict(
+        alert.dict.get(alert.get_key("prv_sources")), _create_source_fields_list(alert)
+    )
+    source_lite_dict = _create_lite_dict(
+        alert.dict.get(alert.get_key("source")), _create_source_fields_list(alert)
+    )
+    alert_lite_dict = {
+        alert.get_key("alertid"): alert.alertid,
+        alert.get_key("source") + "-lite": source_lite_dict,
+        alert.get_key("prv_sources") + "-lite": prev_sources_lite_dict,
+        alert.get_key("object") + "-lite": object_lite_dict,
+    }
+
+    return pittgoogle.Alert.from_dict(payload=alert_lite_dict, schema_name=f"{SURVEY}.lite")
+
+
+def _create_source_fields_list(alert: pittgoogle.Alert) -> list[str]:
+    """Creates a list of survey-specific field names to be included in the lite alert for the source dictionary."""
+    broker_field_names = [
+        "sourceid",
+        "mjd",
+        "ra",
+        "ra_err",
+        "dec",
+        "dec_err",
+        "flux",
+        "flux_err",
+        "filter",
+    ]
+
+    return _get_survey_field_names(alert, broker_field_names)
+
+
+def _create_object_fields_list(alert: pittgoogle.Alert) -> list[str]:
+    """Creates a list of survey-specific field names to be included in the lite alert for the object dictionary."""
+    broker_field_names = [
+        "objectid",
+        "nearby_object1",
+        "nearby_object2",
+        "nearby_object3",
+        "nearby_object1_distance",
+        "nearby_object2_distance",
+        "nearby_object3_distance",
+        "prob_is_object1",
+        "prob_is_object2",
+        "prob_is_object3",
+        "u_flux_err_mean",
+        "g_flux_err_mean",
+        "r_flux_err_mean",
+        "i_flux_err_mean",
+        "z_flux_err_mean",
+        "y_flux_err_mean",
+    ]
+
+    return _get_survey_field_names(alert, broker_field_names)
+
+
+def _get_survey_field_names(alert: pittgoogle.Alert, broker_field_names: list) -> list[str]:
+    """Returns a list of survey-specific field names to be included in the lite dictionary."""
+    _survey_field_names = [alert.get_key(field) for field in broker_field_names]
+
+    # fields may be lists, extract the second element
+    survey_field_names = [
+        _survey_field_name[1] if isinstance(_survey_field_name, list) else _survey_field_name
+        for _survey_field_name in _survey_field_names
+    ]
```

Comment:
What's the reason for extracting the second element? I'm getting a bit lost on what's happening between the different function calls and what the final result will look like.

Reply:
Here's an example:

```
broker_field_names = ["source", "ra", "dec"]
_survey_field_names = [alert.get_key(field) for field in broker_field_names]
print(_survey_field_names)

['diaSource', ['diaSource', 'ra'], ['diaSource', 'dec']]
```

As you can see, `get_key()` may sometimes return a list, so we want to extract the second element from the list.

```
survey_field_names = [
    _survey_field_name[1] if isinstance(_survey_field_name, list) else _survey_field_name
    for _survey_field_name in _survey_field_names
]
print(survey_field_names)

['diaSource', 'ra', 'dec']
```